---
title: "STAG2KD Analyis for A673 and TC71 Cell Lines."
author: "Trent Gutierrez"
date: "`r Sys.Date()`"
output:
  html_document:
    theme: lumen
    code_folding: hide
    toc: yes
    toc_float:
      collapsed: yes
      smooth_scroll: yes
  pdf_document:
    toc: yes
---

```{r Packages, message=FALSE, warning=FALSE, include=FALSE}
library(tidyverse)
library(BiocManager)
library(RColorBrewer)
library(pheatmap)
library(DT)
library(DESeq2)
library(fgsea)
library(dplyr)
library(ggplot2)
library(plotly)
library(styler)
library(kableExtra)
library(fgsea)
library(stats)
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(devtools)
library(conflicted)
library(stringr)
library(fgsea)
library(data.table)
library(biomaRt)
library(EnhancedVolcano)
library(readr)
library(tximport)
library(SummarizedExperiment)
library(pathview)
library(gage)
library(gageData)
library(styler)
library(lintr)
library(ensembldb)
library(EnsDb.Hsapiens.v86)
library(plyr)
library(limma)
library(VennDiagram)
library(msigdbr)
library(ggVennDiagram)

```

```{r include=FALSE, setup,}
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
```

# Introduction

### Research Question

**What is STAG2? What should we expect if we knocked out STAG2 from a cell?**

STAG2 is a important subunit in the cohesin complex that plays an important role in regulating sister chromatid alignment during cell division, and other genetic regulatory functions. Studies show that STAG2-mut EWS have higher rates of metastatic disease and worse outcomes. It is expected that the TC71 & A673 cell lines will have differing phenotype effects to STAG2 KO, for example STAG2 deletion will lead to TC71 growth defect but with A673 it will lead to a growth advantage. STAG1 levels could also increase, in order to possibly fill in for the removal of STAG2 in the cell cycle, a decrease of RAD21 is also observed. The cohesin complex also mediates intrachromosomal interactions including those conjoining enhancers to promoters. Loss of STAG2 produces highly consistent and stable transcriptional changes that may undergo selection to confer a competitive advantage. Two neurodvelopmental transcription factors, POU3F2 and NR2F1 were consistently upregulated in STAG2 KO studies. (Adane et al, 2022)

STAG2 and CDKN24 seem to share an exclusive pattern of genetic alterations. cell lines with STAG2 mutations seem likely to express p16 , and reciprocally all cases with CDKN2A deletion seem to express STAG2. (Tirode et al. 2015)

In a study by CellPress, a GSEA enrichment analysis was conducted and from the 18,889 signatures ranked by the average normalized enrichment score (NES), several of the top 20 signatures enriched in STAG2 proficient condition were EWSR1-FLI1-regulated gene signatures. (Surdez et al. 2021)

The expected pathways for Ewing Sarcoma was also reviewed to compare with the gsea and enrichr results. For a study by JBUON, multiple test were conducted first we'll list is the top 10 significantly up-regulated and down-regulated top: UGT3A2, HMCN1, RBM11, DKK2, SNORA23, PTPN13, TNFAIP6, LIPI, DCC, HOXD10 and down: ATP1B1, CLU, MAOB, SORBS1, SORL1, SYNPO2, KIAA1324, GATM, IGKV2D-28, CKMT1B. and for the GO enrichment analysis, for BP ontology pathways relating to transcription, chromatin modfication and remodeling, SRP-dependent cotranslational protein targeting to membrane, viral transcription, rRNA processing, DNA replication. for the CC enriched pathways, nucleus, nuceloplasm, nucleous, centrosome, cytoplasm, nuclear speck, focal adhesion, nuclear membrane, membrane, nuclear chromatin. For MF ontology the pathways are poly(A) RNA binding, DNA binding, chromatin binding, nucleic acid binding, protein binding, nucleotide binding, helicase binding. (Yan et al. 2018)

### Data input

All samples are downloaded from the SRA Run Selector BioProject PRJNA549593, slight quality control via fastp was conducted and then the reads were aligned using Salmon for Transcript-level quantification files. The metadata for the samples was provided from the SRA Run Selector also. 3 sample comparisons were constructed using the sample data:

1.  SA2 KO vs WT in A673 cells.

2.  SA2 KO vs WT in TC71 cells.

```{r Tximport set up, message=FALSE, warning=FALSE, cache=FALSE}
samples <- read.table(file.path("Analysis/SraRunTable.txt"), sep = ",", header = TRUE) %>%
  dplyr::mutate(cell_line = ifelse(grepl("A673", x = source_name), "A673", "TC71")) %>%
  dplyr::mutate(condition = ifelse(grepl("WT|siCT", x = GENOTYPE), "Control", "Treatment"))

A673_samples <- samples %>%
  dplyr::filter(GENOTYPE %in% c("WT", "SA2 KO") & cell_line == "A673")

TC71_samples <- samples %>%
  dplyr::filter(GENOTYPE %in% c("SA2 KO", "WT") & cell_line == "TC71")

A673_salmon_files <- file.path("Salmon/salmon.out", A673_samples$Run, "quant.sf") %>%
  setNames(object = , A673_samples$Run)

TC71_salmon_files <- file.path("Salmon/salmon.out", TC71_samples$Run, "quant.sf") %>%
  setNames(object = , TC71_samples$Run)

ensdb <- EnsDb.Hsapiens.v86

transcripts <- transcripts(ensdb, columns = c(listColumns(ensdb, "tx"), "gene_name"), return.type = "data.frame") %>%
  as_tibble() %>%
  dplyr::select(tx_id, gene_name)

A673_txi <- tximport(A673_salmon_files, type = "salmon", tx2gene = transcripts, ignoreTxVersion = TRUE)

TC71_txi <- tximport(TC71_salmon_files, type = "salmon", tx2gene = transcripts, ignoreTxVersion = TRUE)

A673_dds_txi <- DESeqDataSetFromTximport(A673_txi, colData = A673_samples, design = ~condition)

TC71_dds_txi <- DESeqDataSetFromTximport(TC71_txi, colData = TC71_samples, design = ~condition)
```

### Collected Sample Metadata

```{r All Samples}
data_input_all <- samples %>%
  dplyr::select(GENOTYPE, cell_line) %>%  
  kbl(caption = "Table 1: Collected Sample Overview") %>%
  kable_styling(bootstrap_options = "striped", full_width = T, html_font = "Cambria")
data_input_all
```

## Exploratory Data

### PCA Plots

```{r A673 PCA Plot, message=FALSE, warning=FALSE}
vst <- vst(A673_dds_txi)

A673_PCA <- plotPCA(vst, intgroup = c("cell_line", "GENOTYPE"), returnData = TRUE)
A673_percentVar <- round(100 * attr(A673_PCA, "percentVar"))

ggplot(A673_PCA, aes(PC1, PC2, color = GENOTYPE)) +
  geom_point(size = 3) +
  ggtitle("PCA Plot for A673 STAG2KO samples") +
  xlab(paste0("PC1: ", A673_percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", A673_percentVar[2], "% variance")) +
  coord_fixed()
```

```{r TC71 PCA plot, message=FALSE, warning=FALSE}
TC71_vst <- vst(TC71_dds_txi)

TC71_PCA <- plotPCA(TC71_vst, intgroup = c("cell_line", "GENOTYPE"), returnData = TRUE)
TC71_percentVar <- round(100 * attr(TC71_PCA, "percentVar"))

ggplot(TC71_PCA, aes(PC1, PC2, color = GENOTYPE)) +
  geom_point(size = 3) +
  ggtitle("PCA Plot for TC71 STAG2KO samples") +
  xlab(paste0("PC1: ", TC71_percentVar[1], "% variance")) +
  ylab(paste0("PC2: ", TC71_percentVar[2], "% variance")) +
  coord_fixed()
```

### Analysis Samples Metadata

```{r metadata, echo=TRUE, message=FALSE, warning=FALSE}
A673_metadata <- colData(A673_dds_txi)
TC71_metadata <- colData(TC71_dds_txi)

sample_metadata <- rbind(A673_metadata, TC71_metadata)

sample_metadata %>%
  as.data.frame() %>%
  dplyr::select(GENOTYPE, cell_line) %>%
  kbl(caption = "Table 2: Analysis Sample Overview") %>%
  kable_styling(bootstrap_options = "striped", full_width = T, html_font = "Cambria")
```

```{r DESeq2, message=FALSE, warning=FALSE, cache=FALSE, include=FALSE}
A673_dds <- DESeq(A673_dds_txi)

TC71_dds <- DESeq(TC71_dds_txi)
```

# Results

```{r A673 result extraction, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}
A673_result <- results(A673_dds, contrast = c("condition", "Treatment", "Control"))

A673_result_df <- as.data.frame(A673_result)

A673_sig_ordered_result <- as.data.frame(A673_result) %>%
  rownames_to_column(var = "Gene_name") %>%
  dplyr::filter(padj < 0.05) %>%
  dplyr::arrange(desc(stat))
```

```{r TC71 result extraction, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}
TC71_result <- results(TC71_dds, contrast = c("condition", "Treatment", "Control"))

TC71_result_df <- as.data.frame(TC71_result)

TC71_sig_ordered_result <- as.data.frame(TC71_result) %>%
  rownames_to_column(var = "Gene_name") %>%
  dplyr::filter(padj < 0.05) %>%
  dplyr::arrange(desc(stat))
```

## Statistical Analysis

### Volcano Plots

```{r A673 EnhancedVolcano plot, warning=FALSE, fig.dim = c(10,8)}
EnhancedVolcano(A673_sig_ordered_result,
  lab = A673_sig_ordered_result$Gene_name,
  x = "log2FoldChange",
  y = "pvalue",
  title = "Siginifcant Genes for STAG2KO in A673 cells",
  subtitle = "",
  pointSize = 1.0,
  labSize = 4.0,
  xlim = c(min(A673_sig_ordered_result$log2FoldChange), max(A673_sig_ordered_result$log2FoldChange)),
  ylim = c(0, 300)
)
```

```{r TC71 EnhancedVolcano plot, warning=FALSE, fig.dim = c(10,8)}
EnhancedVolcano(TC71_sig_ordered_result,
  lab = TC71_sig_ordered_result$Gene_name,
  x = "log2FoldChange",
  y = "pvalue",
  title = "Siginifcant Genes for STAG2KO in TC71 Cells",
  subtitle = "",
  pointSize = 1.0,
  labSize = 4.0,
  xlim = c(min(TC71_sig_ordered_result$log2FoldChange), max(TC71_sig_ordered_result$log2FoldChange)),
  ylim = c(0, 200)
)
```

### Significant DEG tables

```{r A673 significant DEGs Result Table}
A673_table_result <- dplyr::select(A673_sig_ordered_result, Gene_name, log2FoldChange, stat, pvalue, padj)

datatable(A673_table_result,
  class = "cell-border stripe",
  caption = "Table 3: A672 STAG2KO Differentally Significant Genes", rownames = FALSE
)
```

```{r all STAG2KO sample significant DEGs Result Table, message=FALSE, warning=FALSE}
TC71_table_result <- dplyr::select(TC71_sig_ordered_result, Gene_name, log2FoldChange, stat, pvalue, padj)

datatable(TC71_table_result,
  class = "cell-border stripe",
  caption = "Table 4: STAG2KO samples Differentally Significant Genes", rownames = FALSE
)
```

### STAG2 Plot Counts

```{r STAG2 KO plot Counts for A673}
A673_counts <- plotCounts(A673_dds, gene = "STAG2", returnData = TRUE)

ggplot(A673_counts, aes(x =condition, y = count, color = condition)) +
  geom_point(size = 2) +
  ggtitle("Count of STAG2 for A672 SA2KO vs WT")
```

```{r STAG2 KO plot Counts for TC71}
TC71_counts <- plotCounts(TC71_dds, gene = "STAG2", returnData = TRUE)

ggplot(TC71_counts, aes(x =condition, y = count, color = condition)) +
  geom_point(size = 2) +
  ggtitle("Count of STAG2 for TC71 SA2KO vs WT")
```

### Venn Diagram

```{r Venn Diagram Set Up, message=FALSE, warning=FALSE}
A673_overexpressed <- A673_sig_ordered_result %>%
  dplyr::filter(log2FoldChange > 0)
  
A673_underexpressed <- A673_sig_ordered_result %>%
  dplyr::filter(log2FoldChange < 0) 
  
TC71_overexpressed <- TC71_sig_ordered_result %>%
  dplyr::filter(log2FoldChange > 0) 
  
TC71_underexpressed <- TC71_sig_ordered_result %>%
  dplyr::filter(log2FoldChange < 0) 
  
grid.newpage()

over <- venn.diagram(list(A673_overexpressed$Gene_name, TC71_overexpressed$Gene_name),
                     category.names = c("A673","TC71"),
                     filename = NULL,
                     main = "Overlapping Overexpressed Genes",
                     fill = c("red", "deepskyblue4"), 
                     lwd = 1, lty = 1)
grid.draw(over)

grid.newpage()

under <- venn.diagram(list(A673_underexpressed$Gene_name, TC71_underexpressed$Gene_name),
                     category.names = c("A673","TC71"),
                     filename = NULL,
                     main = "Overlapping Underexpressed Genes",
                     fill = c("red", "deepskyblue4"), 
                     lwd = 1, lty = 1)
grid.draw(under)
```

### 4-way plot

```{r 4-way plot, message=FALSE, warning=FALSE}
A673_list <- A673_result_df %>%
  dplyr::select(log2FoldChange, stat, padj) %>%
  rownames_to_column() %>%
  dplyr::rename("Gene" = rowname, "A673_Log2FC" = log2FoldChange, "A673_stat" = stat, "A673_padj" = padj) %>%
  drop_na()


TC71_list <- TC71_result_df %>%
  dplyr::select(log2FoldChange, stat, padj) %>%
  rownames_to_column() %>%
  dplyr::rename("Gene" = rowname, "TC71_Log2FC" = log2FoldChange, "TC71_stat" = stat, "TC71_padj" = padj) %>%
  drop_na()
  
fourway_df <- inner_join(A673_list, TC71_list, by = "Gene") %>%
  dplyr::mutate(Sig_Group = case_when(
    A673_padj < 0.05 & TC71_padj < 0.05 ~ "Both", 
    A673_padj < 0.05 ~ "A673-only", TC71_padj < 0.05 ~ "TC71-only",
    TRUE ~ "Not Significant"
  ))
  
dge_fourway <- ggplot(data = fourway_df, aes(x = A673_stat, y = TC71_stat, label = Gene, color = Sig_Group)) +
  geom_point(alpha = .8) +
  geom_hline(yintercept = 0, size = .1) +
  geom_vline(xintercept = 0, size = .1) +
  coord_fixed(ratio = 1) +
  scale_color_manual(values = c(
    "Both" = "blue",
    "A673-only" = "darkgoldenrod2",
    "TC71-only" = "firebrick",
    "Not Significant" = "grey")) +
  labs(title = "Wald stat of Differentialy Expressed Genes in A673 and TC71 Cell lines")

ggplotly(dge_fourway)
```

### Top 10 Over-expressed and 10 Under-expressed heatmaps

```{r A673 Heatmap Data Setup, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}
A673_top_over <- dplyr::slice_max(A673_sig_ordered_result, n = 10, order_by = stat)
A673_top_under <- dplyr::slice_min(A673_sig_ordered_result, n = 10, order_by = stat)
A673_expression_data <- rbind(A673_top_over, A673_top_under)

A673_normalized_dds_counts <- counts(A673_dds, normalized = TRUE)

A673_sig_norm_dds_counts <- A673_normalized_dds_counts[A673_expression_data$Gene_name, ]

A673_heat_meta <- as.data.frame(colData(A673_dds)) %>%
  dplyr::select(condition, GENOTYPE)

cols_RdBu <- brewer.pal(11, "RdBu")
palette <- colorRampPalette(cols_RdBu)
```

```{r A673 Heatmap plot}
A673_heatmap <- pheatmap(A673_sig_norm_dds_counts,
  main = "Top 10 Over- and Under- expressed A673 STAG2KO DEGs",
  color = palette(200),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_rownames = TRUE,
  annotation = dplyr::select(A673_heat_meta, condition),
  scale = "row"
)
```

```{r TC71 Heatmap Data Setup, message=FALSE, warning=FALSE, cache=TRUE, include=FALSE}
TC71_top_over <- dplyr::slice_max(TC71_sig_ordered_result, n = 10, order_by = stat)
TC71_top_under <- dplyr::slice_min(TC71_sig_ordered_result, n = 10, order_by = stat)
TC71_expression_data <- rbind(TC71_top_over, TC71_top_under)

TC71_normalized_dds_counts <- counts(TC71_dds, normalized = TRUE)

TC71_sig_norm_dds_counts <- TC71_normalized_dds_counts[TC71_expression_data$Gene_name, ]

TC71_heat_meta <- as.data.frame(colData(TC71_dds)) %>%
  dplyr::select(condition, GENOTYPE)
```

```{r TC71 Heatmap plot}
TC71_heatmap <- pheatmap(TC71_sig_norm_dds_counts,
  main = "Top 10 Over- and Under- expressed TC71 STAG2KO DEGs",
  color = palette(200),
  cluster_rows = FALSE,
  cluster_cols = FALSE,
  show_rownames = TRUE,
  annotation = dplyr::select(TC71_heat_meta, condition),
  scale = "row"
)
```

## GSEA Enrichment Analysis

```{r GSEA set up, message=FALSE, warning=FALSE, include=FALSE, cache=TRUE}
gene_sets <- msigdbr(species = "Homo sapiens", category = "C5") %>%
  dplyr::select(gs_name, gene_symbol)

A673_gsea_gene_list <- A673_sig_ordered_result$stat %>%
  setNames(object = , A673_sig_ordered_result$Gene_name) %>%
  sort(decreasing = TRUE)

TC71_gsea_gene_list <- TC71_sig_ordered_result$stat %>%
  setNames(object = , TC71_sig_ordered_result$Gene_name) %>%
  sort(decreasing = TRUE)

A673_gsea_res <- GSEA(geneList = A673_gsea_gene_list, 
                      TERM2GENE = gene_sets)

TC71_gsea_res <- GSEA(geneList = TC71_gsea_gene_list, 
                      TERM2GENE = gene_sets)
```

### GSEA Dot Plot

```{r GSEA dot plot, message=FALSE, warning=FALSE}
dotplot(A673_gsea_res, font.size = 8) +
  ggtitle("A673 GSEA")
```

```{r TC71 GSEA dot plot, message=FALSE, warning=FALSE}
dotplot(TC71_gsea_res, font.size = 8) +
  ggtitle("TC71 GSEA")
```

### GSEA Pathway Plots

```{r A673 Pathway plots}
A673_gsea_df <- as.data.frame(A673_gsea_res)

A673_top_pathways <- A673_gsea_df %>%
  top_n(n = 4, wt = NES) %>%
  pull(ID)
A673_top_pathways_plot <- gseaplot2(A673_gsea_res, geneSetID = A673_top_pathways, title = "Top 4 top A673 Pathways") 

A673_top_pathways_plot

A673_bot_pathways <- A673_gsea_df %>%
  top_n(n = 4, wt = -NES) %>%
  pull(ID)
A673_bot_pathways_plot <- gseaplot2(A673_gsea_res, geneSetID = A673_bot_pathways, title = "Top 4 bottom A673 Pathways") 

A673_bot_pathways_plot
```

```{r TC71 Pathway plots}
TC71_gsea_df <- as.data.frame(TC71_gsea_res)

TC71_top_pathways <- TC71_gsea_df %>%
  top_n(n = 4, wt = NES) %>%
  pull(ID)
TC71_top_pathways_plot <- gseaplot2(TC71_gsea_res, geneSetID = TC71_top_pathways, title = "Top 4 top TC71 Pathways") 

TC71_top_pathways_plot

TC71_bot_pathways <- TC71_gsea_df %>%
  top_n(n = 4, wt = -NES) %>%
  pull(ID)
TC71_bot_pathways_plot <- gseaplot2(TC71_gsea_res, geneSetID = TC71_bot_pathways, title = "Top 4 bottom TC71 Pathways") 

TC71_bot_pathways_plot
```

### GSEA Venn Diagram

```{r GSEA Venn Diagram}
pathways <- ggVennDiagram(x = list(A673 = A673_gsea_df$ID, TC71 = TC71_gsea_df$ID)) +
  ggtitle("Overlapping Pathways between Cell Lines") +
  theme(plot.title = element_text(hjust = 0.5)) +
  scale_color_brewer(palette = "Blues") +
  scale_fill_gradient(low = "#ece7f2", high = "#2b8cbe")
pathways
```

### GSEA Pathway Datatables

```{r GSEA Venn Diagram Tables}
path_overlap <- intersect(A673_gsea_df$ID, TC71_gsea_df$ID) 
"%!in%" <- Negate("%in%")

cell_gsea_list <- list('A673' = A673_gsea_df, "TC71" = TC71_gsea_df)

pathway_data <- lapply(cell_gsea_list, function(x){ (x[x$ID %!in% path_overlap,] %>%
                                                       rownames_to_column("Pathway") %>%
                                                       dplyr::arrange(rank) %>%
                                                       dplyr::select(c(-ID, -Description, -setSize,
                                                                       -enrichmentScore, -qvalue, -leading_edge,
                                                                       -core_enrichment)) %>%
                                                       dplyr::arrange(desc(NES)))
  })

pathway_data$A673 %>%
  datatable(caption = "Table 5: A673 GSEA Pathways")

pathway_data$TC71 %>%
  datatable(caption = "Table 6: TC71 GSEA Pathways")
  
both_pathways <- inner_join(A673_gsea_df, TC71_gsea_df, by = "ID") %>%
  dplyr::mutate(meanrank = rowMeans(cbind(rank.x, rank.y))) %>%
  dplyr::arrange(meanrank) %>%
  dplyr::select(starts_with(c("ID", "NES", "p.adjust")))

names(both_pathways) <- gsub("*.y", ".TC71", names(both_pathways))
names(both_pathways) <- gsub("*.x", ".A673", names(both_pathways))

datatable(both_pathways, caption = "Table 7: Shared Pathways for Both Cell Lines")
```

## Enrichr Pathway Analysis

```{r A673 enrichr set up, include=FALSE}
A673_gene_list <- A673_sig_ordered_result %>%
  dplyr::mutate(result = case_when(
    log2FoldChange > 0 ~ "Over-expressed",
    TRUE ~ "Under-expressed"
  )) %>%
  group_by(result) %>%
  {
    setNames(group_split(.), group_keys(.)[[1]])
  } %>%
  llply(pull, var = Gene_name)
```

### A673 enrichr Pathway enrichment {.tabset}

```{r A673 enrichr API set up, results='asis'}
A673_resRmd <- llply(names(A673_gene_list), function(groupNow) {
  genesNow <- A673_gene_list[[groupNow]]
  response <- httr::POST(
    url = "https://maayanlab.cloud/Enrichr/addList",
    body = list(
      "list" = paste0(genesNow, collapse = "\n"),
      "description" = groupNow
    )
  )
  response <- jsonlite::fromJSON(httr::content(response, as = "text"))
  permalink <- paste0(
    "https://maayanlab.cloud/Enrichr/enrich?dataset=",
    response$shortId[1]
  )
  knitr::knit_child(
    text = c(
      "#### `r groupNow`",
      "",
      'Enrichr Link: <a href="`r permalink`" target="_blank">`r groupNow`</a>.',
      ""
    ),
    envir = environment(),
    quiet = TRUE
  )
})
cat(unlist(A673_resRmd), sep = "\n")
```

```{r TC71 enrichr set up, include=FALSE}
TC71_gene_list <- TC71_sig_ordered_result %>%
  dplyr::mutate(result = case_when(
    log2FoldChange > 0 ~ "Over-expressed",
    TRUE ~ "Under-expressed"
  )) %>%
  group_by(result) %>%
  {
    setNames(group_split(.), group_keys(.)[[1]])
  } %>%
  llply(pull, var = Gene_name)
```

### TC71 enrichr Pathway enrichment {.tabset}

```{r TC71 enrichr API set up, results='asis'}
TC71_resRmd <- llply(names(TC71_gene_list), function(groupNow) {
  genesNow <- TC71_gene_list[[groupNow]]
  response <- httr::POST(
    url = "https://maayanlab.cloud/Enrichr/addList",
    body = list(
      "list" = paste0(genesNow, collapse = "\n"),
      "description" = groupNow
    )
  )
  response <- jsonlite::fromJSON(httr::content(response, as = "text"))
  permalink <- paste0(
    "https://maayanlab.cloud/Enrichr/enrich?dataset=",
    response$shortId[1]
  )
  knitr::knit_child(
    text = c(
      "#### `r groupNow`",
      "",
      'Enrichr Link: <a href="`r permalink`" target="_blank">`r groupNow`</a>.',
      ""
    ),
    envir = environment(),
    quiet = TRUE
  )
})
cat(unlist(TC71_resRmd), sep = "\n")
```

# Discussion

**How do we know that the STAG2 KO treatment worked?**

The PlotCount graphs for our cell lines show that the WT controls do express higher counts of STAG2, however the intensity of the change is different. The TC71 treatment cells had a far more extreme decrease in STAG2 levels compared to a relatively light change in A673 treatment cells.

**What SA2 KO DEGs are specific to TC71? A673? and which are shared between the two?**

The Over- and Under- expressed venn diagrams give us a broad idea of what genes are specific to their cell line and shared between the two. The 4-way plot gives us a closer look at the genes in the samples, we can see what genes are over- and under - expressed for both cell lines and specific cell lines. The genes on the edge of the graph match up to genes listed on the heatmaps.

**Any Notable results?**

Looking at the GSEA pathway results we do not see many shared enriched pathways with only 8% of the pathways being shared. Looking at the shared pathway table the NES of the cell lines are relatively similar doesn't seem to have any major discrepancies.

## Works Cited

Adane B, Alexe G, Seong BKA, Lu D, Hwang EE, Hnisz D, Lareau CA, Ross L, Lin S, Dela Cruz FS, Richardson , Weintraub AS, Wang S, Iniguez AB, Dharia NV, Conway AS, Robichaud AL, Tanenbaum B, Krill-Burger JM, Vazquez F, Schenone M, Berman JN, Kung AL, Carr SA, Aryee MJ, Young RA, Crompton BD, Stegmaier K. STAG2 loss rewires oncogenic and developmental programs to promote metastasis in Ewing sarcoma. Cancer Cell. 2021 Jun 14;39(6):827-844.e10. doi: 10.1016/j.ccell.2021.05.007. PMID: 34129824; PMCID: PMC8378827.

Li G, Zhang P, Zhang W, Lei Z, He J, Meng J, Di T, Yan W. Identification of key genes and pathways in Ewing's sarcoma patients associated with metastasis and poor prognosis. Onco Targets Ther. 2019 May 27;12:4153-4165. doi: 10.2147/OTT.S195675. PMID: 31213834; PMCID: PMC6549663.

Surdez D, Zaidi S, Grossetête S, Laud-Duval K, Ferre AS, Mous L, Vourc'h T, Tirode F, Pierron G, Raynal V, Baulande S, Brunet E, Hill V, Delattre O. STAG2 mutations alter CTCF-anchored loop extrusion, reduce cis-regulatory interactions and EWSR1-FLI1 activity in Ewing sarcoma. Cancer Cell. 2021 Jun 14;39(6):810-826.e9. doi: 10.1016/j.ccell.2021.04.001. Epub 2021 Apr 29. PMID: 33930311.

Tirode F, Surdez D, Ma X, Parker M, Le Deley MC, Bahrami A, Zhang Z, Lapouble E, Grossetête-Lalami S, Rusch M, Reynaud S, Rio-Frio T, Hedlund E, Wu G, Chen X, Pierron G, Oberlin O, Zaidi S, Lemmon G, Gupta P, Vadodaria B, Easton J, Gut M, Ding L, Mardis ER, Wilson RK, Shurtleff S, Laurence V, Michon J, Marec-Bérard P, Gut I, Downing J, Dyer M, Zhang J, Delattre O; St. Jude Children's Research Hospital--Washington University Pediatric Cancer Genome Project and the International Cancer Genome Consortium. Genomic landscape of Ewing sarcoma defines an aggressive subtype with co-association of STAG2 and TP53 mutations. Cancer Discov. 2014 Nov;4(11):1342-53. doi: 10.1158/2159-8290.CD-14-0622. Epub 2014 Sep 15. PMID: 25223734; PMCID: PMC4264969.

Yan C, Wang Y, Wang Q, Feng X, Wang L, Bu Z, Lu B, Jiang J. Identification of key genes and pathways in Ewing's sarcoma using bioinformatics analysis. J BUON. 2018 Sep-Oct;23(5):1472-1480. PMID: 30570875.
